src/messages/MOSDFastDispatchOp.h

```diff
@@ -0,0 +1,19 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_MOSDFASTDISPATCHOP_H
+#define CEPH_MOSDFASTDISPATCHOP_H
+
+#include "msg/Message.h"
+#include "osd/osd_types.h"
+
+class MOSDFastDispatchOp : public Message {
+public:
+  virtual epoch_t get_map_epoch() const = 0;
+  virtual spg_t get_spg() const = 0;
```

Comment:
This subclassing should also let us remove the templated replica_op_required_epoch() and handle_replica_op() functions in OSD.